File: GEMA_andorid/GEMAINVENTORY/app/src/main/java/com/example/gemainventory/model/ProductoDto.java
package com.example.gemainventory.model;

import com.google.gson.annotations.SerializedName;

public class ProductoDto {

    @SerializedName("idProducto")
    private String idProducto;

    @SerializedName("nombre")
    private String nombre;

    @SerializedName("sku")
    private String sku;

    @SerializedName("cantidad")
    private Integer cantidad;

    @SerializedName("categoria")
    private String categoria;

    @SerializedName("precioCompra")
    private Double precioCompra;

    @SerializedName("precioVenta")
    private Double precioVenta;

    @SerializedName("descripcion")
    private String descripcion;

    @SerializedName("stockMinimo")
    private Integer stockMinimo;

    @SerializedName("idUsuario")
    private String idUsuario;

    @SerializedName("idAlmacen")
    private Integer idAlmacen;


    public ProductoDto(String nombre, String sku, Integer cantidad, String categoria,
                       Double precioCompra, Double precioVenta, String descripcion, Integer stockMinimo, String idUsuario, Integer idAlmacen) {
        this.nombre = nombre;
        this.sku = sku;
        this.cantidad = cantidad;
        this.categoria = categoria;
        this.precioCompra = precioCompra;
        this.precioVenta = precioVenta;
        this.descripcion = descripcion;
        this.stockMinimo = stockMinimo;
        this.idUsuario = idUsuario;
        this.idAlmacen = idAlmacen;
    }

    public String getNombre() { return nombre; }
    public void setNombre(String nombre) { this.nombre = nombre; }

    public String getIdProducto() { return idProducto; }

    // Campos Multimedia
    @SerializedName("imagenUrl")
    private String imagenUrl;
    @SerializedName("modelo3dUrl")
    private String modelo3dUrl;

    public String getImagenUrl() { return imagenUrl; }
    public void setImagenUrl(String imagenUrl) { this.imagenUrl = imagenUrl; }
    public String getModelo3dUrl() { return modelo3dUrl; }
    public void setModelo3dUrl(String modelo3dUrl) { this.modelo3dUrl = modelo3dUrl; }

    public Double getPrecioVenta() { return precioVenta; }

    public String getSku() { return sku; }
    public Integer getCantidad() { return cantidad; }
    public String getCategoria() { return categoria; }
    public Double getPrecioCompra() { return precioCompra; }
    public String getDescripcion() { return descripcion; }
    public Integer getStockMinimo() { return stockMinimo; }
    public String getIdUsuario() { return idUsuario; }
    public Integer getIdAlmacen() { return idAlmacen; }
    public void setIdAlmacen(Integer idAlmacen) { this.idAlmacen = idAlmacen; }

    @Override
    public String toString() {
        if (nombre == null) return "Sin Nombre (Error de Conexión)";
        return nombre;
    }
}

File: GEMA_andorid/GEMAINVENTORY/app/src/main/java/com/example/gemainventory/model/ProductoDto.kt
package com.example.gemainventory.model

import com.google.gson.annotations.SerializedName

data class ProductoDto(
    @SerializedName("nombre")
    var nombre: String = "",

    @SerializedName("sku")
    var sku: String = "",

    @SerializedName("cantidad")
    var cantidad: Int = 0,

    @SerializedName("categoria")
    var categoria: String = "",

    @SerializedName("precioCompra")
    var precioCompra: Double = 0.0,

    @SerializedName("precioVenta")
    var precioVenta: Double = 0.0,

    @SerializedName("descripcion")
    var descripcion: String = "",

    @SerializedName("stockMinimo")
    var stockMinimo: Int = 5,

    @SerializedName("idUsuario")
    var usuarioId: String = "",

    @SerializedName(value = "id_almacen", alternate = ["idAlmacen", "almacen_id", "almacenId"])
    var idAlmacen: Int? = null,

    @SerializedName("nombreAlmacen")
    var nombreAlmacen: String? = null,

    @SerializedName("idProducto")
    val idProducto: String? = null,

    @SerializedName("imagenUrl")
    var imagenUrl: String? = null,

    @SerializedName("modelo3dUrl")
    var modelo3dUrl: String? = null
) {
    override fun toString(): String {
        return nombre.ifEmpty { "Sin Nombre" }
    }

    fun toCreacionDto(userId: String): ProductoCreacionDto {
        val dto = ProductoCreacionDto(
            nombre,
            sku,
            cantidad,
            categoria,
            precioCompra,
            precioVenta,
            descripcion,
            stockMinimo,
            userId,
            idAlmacen
        )
        dto.imagenUrl = imagenUrl
        dto.modelo3dUrl = modelo3dUrl
        return dto
    }
}
